Mathlib/LinearAlgebra/Matrix/NonsingularInverse.lean

```diff
@@ -483,6 +483,43 @@ theorem inv_eq_left_inv (h : B * A = 1) : A⁻¹ = B :=
 theorem inv_eq_right_inv (h : A * B = 1) : A⁻¹ = B :=
   inv_eq_left_inv (mul_eq_one_comm.2 h)
 
+section Woodbury
+
+variable [Fintype m] [DecidableEq m]
+variable (A: Matrix n n α) (U : Matrix n m α) (C : Matrix m m α) (V : Matrix m n α)
+variable [Invertible A] [Invertible C] [Invertible (⅟C + V * ⅟A* U)]
+
+lemma add_mul_mul_invOf_mul_eq_one : (A + U*C*V)*(⅟A - ⅟A*U*⅟(⅟C + V*⅟A*U)*V*⅟A) = 1 := by
```

Comment:
Can you move this to `Matrix/Invertible.lean`, and generalize it from `CommRing` to `Ring`?

Reply:
Should i move all of it or just this lemma?

Reply:
This lemma and the `Invertible` def. The lemma I suggest below about `inv` would remain in this file.

Reply:
@4hma4d, do you want me to do the move for you? Happy to do so to get this PR over the line, but don't want to tread on your toes.

Reply:
Sorry for the delay. I didn't move `add_mul_mul_inv` since it relies on `Matrix.invertibleOfRightInverse` which is proved in `NonsingularInverse`

Reply:
> I didn't move `add_mul_mul_inv` since it relies on `Matrix.invertibleOfRightInverse` which is proved in `NonsingularInverse`

Indeed, to prove `add_mul_mul_inv` for the non-commutative case you have to repeat the whole proof, but with multiplication on the left instead! I've pushed a commit that does that.